### ISSUE TITLE
[BUG/ISSUE] Parallelization error in TPCORE_FVDAS (i.e. GEOS-Chem Classic transport module)

### ISSUE THREAD

**Issue by yantosca** [geoschem/geos-chem#1082]
### What institution are you from?
GCST

### Description of the problem
In this parallel loop in routine `TPCORE_FVDAS` (in `GeosCore/tpcore_fvdas_mod.F90`), the variable `DQ1` should be held `!$OMP PRIVATE` but is not:

https://github.com/geoschem/geos-chem/blob/9ab6931db5b556eafd0e002e9fd4fdbcbd5fbb44/GeosCore/tpcore_fvdas_mod.F90#L782-L946

The reason why it DQ1 should be held `!$OMP PRIVATE` is because it is a 3D array that is rewritten on each iteration of IQ.

### Symptoms
This bug causes small numerical differences in output when comparing GEOS-Chem Classic runs that were done on a different number of cores.  If the same number of cores is always used (e.g. for benchmarks) then the parallelization is done in the same order and identical results are obtained.

### Solution
I have rewritten the loop accordingly (while also cleaning up the code a bit and adding extra whitespace etc).  See commit 933adebf3bfedf413a65f9afa607d9c3d9c1b91d.

I fixed this in the development branch for Hg chemistry via KPP (PR #1063), and will bring this in along with the Hg chemistry updates.  I am also testing the impact this has on the fullchem simulation and will post results here.

**Comment by yantosca** [comment]
After this fix we now obtain identical results w/r/t the prior state of the code,  A short difference test with the Hg simulations shows:

```console
Using configuration file hg.yml
... Printing totals and differences
Variable               Ref=alpha.11             Dev=tpcore_fix           Dev - Ref
AREA                 : 510065600000000.0      | 510065600000000.0      | 0.0 
Met_PS1DRY           : 3193313.8              | 3193313.8              | 0.0 
Met_PS1WET           : 3199046.2              | 3199046.2              | 0.0 
Met_DELPDRY          : 3191367.0              | 3191367.0              | 0.0 
Met_SPHU1            : 305599.25              | 305599.25              | 0.0 
Met_TMPU1            : 57776276.0             | 57776276.0             | 0.0 
SpeciesRst_Br        : 1.9540794e-15          | 1.9540794e-15          | 0.0 
SpeciesRst_BrO       : 1.9540794e-15          | 1.9540794e-15          | 0.0 
SpeciesRst_CH4       : 1.9540794e-15          | 1.9540794e-15          | 0.0 
SpeciesRst_CO        : 1.9540794e-15          | 1.9540794e-15          | 0.0 
SpeciesRst_Cl        : 1.9540794e-15          | 1.9540794e-15          | 0.0 
SpeciesRst_ClO       : 1.9540794e-15          | 1.9540794e-15          | 0.0 
SpeciesRst_HO2       : 7.816322e-10           | 7.816322e-10           | 0.0 
SpeciesRst_Hg0       : 3.4637026e-11          | 3.4637026e-11          | 0.0 
SpeciesRst_Hg2ClP    : 1.3432364e-11          | 1.3432364e-11          | 0.0 
SpeciesRst_Hg2ORGP   : 3.235946e-13           | 3.235946e-13           | 0.0 
SpeciesRst_Hg2STRP   : 2.1128466e-12          | 2.1128466e-12          | 0.0 
SpeciesRst_HgBr      : 2.3412479e-13          | 2.3412479e-13          | 0.0 
SpeciesRst_HgBr2     : 2.2195416e-12          | 2.2195416e-12          | 0.0 
SpeciesRst_HgBrBrO   : 1.7767237e-11          | 1.7767237e-11          | 0.0 
SpeciesRst_HgBrClO   : 1.9590347e-13          | 1.9590347e-13          | 0.0 
SpeciesRst_HgBrHO2   : 1.559005e-13           | 1.559005e-13           | 0.0 
SpeciesRst_HgBrNO2   : 2.2940395e-13          | 2.2940395e-13          | 0.0 
SpeciesRst_HgBrO     : 1.9540794e-15          | 1.9540794e-15          | 0.0 
SpeciesRst_HgBrOH    : 2.4927754e-13          | 2.4927754e-13          | 0.0 
SpeciesRst_HgCl      : 3.176558e-13           | 3.176558e-13           | 0.0 
SpeciesRst_HgCl2     : 5.6047116e-13          | 5.6047116e-13          | 0.0 
SpeciesRst_HgClBr    : 2.114469e-11           | 2.114469e-11           | 0.0 
SpeciesRst_HgClBrO   : 2.2337376e-13          | 2.2337376e-13          | 0.0 
SpeciesRst_HgClClO   : 2.2620446e-13          | 2.2620446e-13          | 0.0 
SpeciesRst_HgClHO2   : 2.4881578e-11          | 2.4881578e-11          | 0.0 
SpeciesRst_HgClNO2   : 1.7335565e-13          | 1.7335565e-13          | 0.0 
SpeciesRst_HgClO     : 1.9540794e-15          | 1.9540794e-15          | 0.0 
SpeciesRst_HgClOH    : 2.570505e-13           | 2.570505e-13           | 0.0 
SpeciesRst_HgOH      : 3.726408e-13           | 3.726408e-13           | 0.0 
SpeciesRst_HgOHBrO   : 2.131467e-11           | 2.131467e-11           | 0.0 
SpeciesRst_HgOHClO   : 2.7568803e-13          | 2.7568803e-13          | 0.0 
SpeciesRst_HgOHHO2   : 2.5951225e-13          | 2.5951225e-13          | 0.0 
SpeciesRst_HgOHNO2   : 2.5391269e-11          | 2.5391269e-11          | 0.0 
SpeciesRst_HgOHO     : 1.9540794e-15          | 1.9540794e-15          | 0.0 
SpeciesRst_HgOHOH    : 2.7720472e-13          | 2.7720472e-13          | 0.0 
SpeciesRst_NO        : 7.8163204e-08          | 7.8163204e-08          | 0.0 
SpeciesRst_NO2       : 7.8163204e-08          | 7.8163204e-08          | 0.0 
SpeciesRst_O3        : 0.0039081587           | 0.0039081587           | 0.0 
SpeciesRst_OH        : 7.816322e-10           | 7.816322e-10           | 0.0 
```

**Comment by yantosca** [comment]
We can close out this issue as it has been merged with the 13.4.0 development stream.

**Comment by yantosca** [comment]
Remaining parallelization issues have been solved in PR #1190.
